Core/HostVersionProviders/BuildMap.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using log4net;
+using Newtonsoft.Json;
+
+namespace CKAN.HostVersionProviders
+{
+    // ReSharper disable once ClassNeverInstantiated.Global
+    public sealed class BuildMap : IBuildMap
+    {
+        // TODO: Need a way for the client to configure this
+        private static readonly Uri BuildMapUri =
+            new Uri("https://raw.githubusercontent.com/KSP-CKAN/CKAN-meta/master/builds.json");
+
+        // TODO: Get this through dependency injection
+        private readonly ILog _log = LogManager.GetLogger(typeof(BuildMap));
+
+        private readonly object _buildMapLock = new object();
+        private JBuilds _buildMap;
+
+        private readonly IWin32Registry _registry;
+        
+        public KSPVersion this[string buildId]
```

Comment:
@pjf Ah, yes, [indexers](https://msdn.microsoft.com/en-us/library/6x16t2tx.aspx) are totally a thing. It's why you can do `dictionary["foo"]`. They can even take multiple parameters: `someType["foo", "bar"]`.